polling/nf_configuration.go

```diff
@@ -0,0 +1,124 @@
+// SPDX-FileCopyrightText: 2025 Canonical Ltd
+
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package polling
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/mohae/deepcopy"
+	"github.com/omec-project/openapi/nfConfigApi"
+	"github.com/omec-project/smf/logger"
+)
+
+const (
+	initialPollingInterval = 5 * time.Second
+	pollingMaxBackoff      = 40 * time.Second
+	pollingBackoffFactor   = 2
+	pollingPath            = "/nfconfig/session-management"
+)
+
+type nfConfigPoller struct {
+	currentSessionManagementConfig []nfConfigApi.SessionManagement
+	client                         *http.Client
+}
+
+// StartPollingService initializes the polling service and starts it.
+// The polling service continuously makes HTTP GET request to the webconsole and updates the network configuration
+func StartPollingService(ctx context.Context, webuiUri string, onUpdate func([]nfConfigApi.SessionManagement)) {
+	poller := nfConfigPoller{
+		currentSessionManagementConfig: []nfConfigApi.SessionManagement{},
+		client:                         &http.Client{Timeout: initialPollingInterval},
+	}
+
+	interval := initialPollingInterval
+	pollingEndpoint := webuiUri + pollingPath
+	logger.PollConfigLog.Infof("Started polling service on %s every %v", pollingEndpoint, initialPollingInterval)
+
+	for {
+		select {
+		case <-ctx.Done():
+			logger.PollConfigLog.Infoln("Polling service shutting down")
```

Comment:
Similar comment for the other loggings
```suggestion
			logger.PollConfigLog.Infoln("polling service shutting down")
```

Reply:
Done